File: specifikacija/json2md.py
import json
from pathlib import Path

def json_to_md(json_file):
    path = Path(json_file)
    md_file = path.with_suffix(".md")

    with open(json_file, encoding="utf-8") as f:
        obj = json.load(f)

    lines = [f"# Verzija {obj['version']} ({obj['date']})"]

    if "description" in obj:
        lines.append("")
        lines.append(f"## {obj['description']}")

    lines.append("")

    if not obj["data"]:
        raise ValueError(f"No data found in JSON: {json_file}")

    headers = list(obj["data"][0].keys())

    col_widths = [len(h) for h in headers]
    for row in obj["data"]:
        for i, h in enumerate(headers):
            col_widths[i] = max(col_widths[i], len(str(row.get(h, ""))))

    def pad_cell(text, width):
        return text + " " * (width - len(text))

    header_row = "| " + " | ".join(pad_cell(h, col_widths[i]) for i, h in enumerate(headers)) + " |"
    lines.append(header_row)

    sep_row = "|" + "|".join("-" * max(4, col_widths[i] + 2) for i in range(len(headers))) + "|"
    lines.append(sep_row)

    for row in obj["data"]:
        row_line = "| " + " | ".join(pad_cell(str(row.get(h, "")), col_widths[i]) for i, h in enumerate(headers)) + " |"
        lines.append(row_line)

    with open(md_file, "w", encoding="utf-8") as f:
        f.write("\n".join(lines) + "\n")

def process_dirs():
    base_dir = Path(__file__).parent
    for subdir in ["eracun", "as4"]:
        dir_path = base_dir / subdir
        if dir_path.exists() and dir_path.is_dir():
            for json_file in dir_path.glob("*.json"):
                try:
                    json_to_md(json_file)
                    print(f"Converted: {json_file}")
                except Exception as e:
                    print(f"Failed: {json_file} -> {e}")

if __name__ == "__main__":
    process_dirs()

File: .build/assets/json2md.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

function jsonToMd(jsonFile) {
    const jsonPath = path.resolve(jsonFile);
    const mdFile = jsonPath.replace(/\.json$/, '.md');

    let obj;
    try {
        const content = fs.readFileSync(jsonPath, 'utf8');
        obj = JSON.parse(content);
    } catch (error) {
        throw new Error(`Failed to read JSON file: ${error.message}`);
    }

    const lines = [`# Verzija ${obj.version} (${obj.date})`];

    if (obj.description) {
        lines.push('');
        lines.push(`## ${obj.description}`);
    }

    lines.push('');

    if (!obj.data || obj.data.length === 0) {
        throw new Error(`No data found in JSON: ${jsonFile}`);
    }

    const headers = Object.keys(obj.data[0]);

    // Calculate column widths
    const colWidths = headers.map(h => h.length);
    for (const row of obj.data) {
        headers.forEach((h, i) => {
            const value = row[h] !== undefined ? String(row[h]) : '';
            colWidths[i] = Math.max(colWidths[i], value.length);
        });
    }

    function padCell(text, width) {
        return text + ' '.repeat(width - text.length);
    }

    // Header row
    const headerRow = '| ' + headers.map((h, i) => padCell(h, colWidths[i])).join(' | ') + ' |';
    lines.push(headerRow);

    // Separator row
    const sepRow = '|' + colWidths.map(w => '-'.repeat(Math.max(4, w + 2))).join('|') + '|';
    lines.push(sepRow);

    // Data rows
    for (const row of obj.data) {
        const rowLine = '| ' + headers.map((h, i) => {
            const value = row[h] !== undefined ? String(row[h]) : '';
            return padCell(value, colWidths[i]);
        }).join(' | ') + ' |';
        lines.push(rowLine);
    }

    fs.writeFileSync(mdFile, lines.join('\n') + '\n', 'utf8');
}

function processDirs() {
    const baseDir = path.join(__dirname, '..', '..', 'specifikacija');
    const subdirs = ['eracun', 'as4'];

    for (const subdir of subdirs) {
        const dirPath = path.join(baseDir, subdir);

        if (!fs.existsSync(dirPath)) continue;

        const stat = fs.statSync(dirPath);
        if (!stat.isDirectory()) continue;

        const files = fs.readdirSync(dirPath);
        for (const file of files) {
            if (!file.endsWith('.json')) continue;

            const jsonFile = path.join(dirPath, file);
            try {
                jsonToMd(jsonFile);
                console.log(`Converted: ${jsonFile}`);
            } catch (error) {
                console.error(`Failed: ${jsonFile} -> ${error.message}`);
            }
        }
    }
}

if (require.main === module) {
    processDirs();
}
